src/php/bin/run_tests.sh

```diff
@@ -22,9 +22,18 @@ cd src/php/bin
 source ./determine_extension_dir.sh
 # in some jenkins macos machine, somehow the PHP build script can't find libgrpc.dylib
 export DYLD_LIBRARY_PATH=$root/libs/$CONFIG
+
 php $extension_dir -d max_execution_time=300 $(which phpunit) -v --debug \
   --exclude-group persistent_list_bound_tests ../tests/unit_tests
 
 php $extension_dir -d max_execution_time=300 $(which phpunit) -v --debug \
   ../tests/unit_tests/PersistentChannelTests
 
+export ZEND_DONT_UNLOAD_MODULES=1
+export USE_ZEND_ALLOC=0
+# Detect whether valgrind is executable
+if [ -x "$(command -v valgrind)" ]; then
+  valgrind --error-exitcode=10 --leak-check=yes php $extension_dir -d max_execution_time=300 \
+	../tests/MemoryLeakTest/MemoryLeakTest.php
+  exit 0
```

Comment:
One more nit: we don't need `exit 0` here now. We can let the script finish with whatever status the `valgrind` command finishes with.